jax/_src/ad_util.py

```diff
@@ -76,11 +77,18 @@ def _stop_gradient_impl(x: T) -> T:
   if not valid_jaxtype(x):
     raise TypeError("stop_gradient only works on valid JAX arrays, but "
                     f"input argument is: {x}")
+  a = get_aval(x)
+  if isinstance(a, AbstractRef):
+    raise TypeError(f"can't apply stop_gradient to mutable array, but got {a}")
```

Comment:
Maybe you can also say to apply it's value instead? So `stop_gradient(x[...])`? Is that allowed?

Reply:
It is allowed, but I'm not sure if that's always the solution.

Reply:
I'll add: "Maybe you want to apply it to the value instead, e.g. `x_ref[...]`?"